src/components/Editor/ClientCursor.ts

```diff
@@ -1,88 +1,109 @@
-const delay = 2;
-const SECOND = 1000;
-const REMOVE_TIME = (duration + delay) * SECOND;
+type actorId = string; // used in yorkie-sdk-js client
+type nameRemoveTimerID = ReturnType<typeof setTimeout>;
 
 // REF https://github.com/FujitsuLaboratories/cattaz/blob/master/src/AppEnabledWikiEditorCodeMirror.jsx#L24
 class ClientCursor {
-  id: string;
+  id: actorId;
   color: string;
+  height: number;
+
   marker: CodeMirror.TextMarker | null;
   lineMarker: CodeMirror.TextMarker | null;
 
-  constructor(id: string, color: string) {
+  // Values to display the name on the cursor.
+  nameRemoveTimeMap: Map<actorId, nameRemoveTimerID>;
+  nameAnimationDelay: number;
+  nameAnimationDuration: number;
+  nameRemoveTime: number;
+
+  constructor(id: actorId, color: string) {
     this.id = id;
     this.color = color;
+    this.height = 0;
     this.marker = null;
     this.lineMarker = null;
-  }
 
-  static of(id: string, color: string) {
-    return new ClientCursor(id, color);
+    this.nameRemoveTimeMap = new Map();
+    this.nameAnimationDelay = 1;
+    this.nameAnimationDuration = 0.2;
+    this.nameRemoveTime = (this.nameAnimationDuration + this.nameAnimationDelay) * 1000;
   }
 
   updateCursor(cm: CodeMirror.Editor, cursorPos: CodeMirror.Position) {
     this.removeCursor();
     const cursorCoords = cm.cursorCoords(cursorPos);
-    const cursorElement = document.createElement('span');
-    const size = cursorCoords.bottom - cursorCoords.top;
-
-    cursorElement.style.position = 'absolute';
-    cursorElement.style.borderLeftStyle = 'solid';
-    cursorElement.style.borderLeftWidth = '2px';
-    cursorElement.style.borderLeftColor = this.color;
-    cursorElement.style.height = `${size}px`;
-    cursorElement.style.padding = '0px';
-    cursorElement.style.zIndex = '0';
-
-    const nameBoxElement = document.createElement('span');
-    nameBoxElement.style.position = 'absolute';
-    nameBoxElement.style.top = `-${size}px`;
-
-    const nameElement = document.createElement('span');
-    nameElement.textContent = this.id;
-    nameElement.style.position = 'fixed';
-    nameElement.style.backgroundColor = this.color;
-    nameElement.style.padding = '1px 4px';
-    nameElement.style.borderRadius = '4px';
-    nameElement.style.color = invert(this.color, true);
-    nameElement.style.animationDuration = `${duration}s`;
-    nameElement.style.animationDelay = `${delay}s`;
-    nameElement.className = 'text-remove';
-
-    nameBoxElement.appendChild(nameElement);
-    cursorElement.appendChild(nameBoxElement);
-
-    this.removeNameReserve(nameBoxElement);
+    const cursorEl = document.createElement('span');
+    this.height = cursorCoords.bottom - cursorCoords.top;
+
+    cursorEl.style.position = 'absolute';
+    cursorEl.style.borderLeftStyle = 'solid';
+    cursorEl.style.borderLeftWidth = '2.5px';
+    cursorEl.style.borderLeftColor = this.color;
+    cursorEl.style.height = `${this.height}px`;
+    cursorEl.style.padding = '0px';
```

Comment:
Except for variable value like `this.height`, how about creating a new css class in the css file and connect with this cursorElement?

ex)
```
cursorEl.classList.add('cursor-class')
cursorEl.style.height = `${this.height}px`
```

Reply:
Thank you for your review
It would be better to follow your suggestion!